app/Console/Commands/ImportNasspGroupsCommand.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace Rogue\Console\Commands;
+
+use Exception;
+use League\Csv\Reader;
+use Rogue\Models\Group;
+use Rogue\Models\GroupType;
+use Illuminate\Console\Command;
+
+class ImportNasspGroupsCommand extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'rogue:nassp-groups-import {path}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Create National Association of Secondary School Principals group types and groups from a CSV.';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Returns a trimmed CSV value, or null if equal to the string "NULL".
+     *
+     * @param string
+     * @return string
+     */
+    public function sanitize($value)
+    {
+        $result = trim($value);
+
+        return $result == 'NULL' ? null : $value;
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle()
+    {
+        // Sample CSV path: https://gist.githubusercontent.com/aaronschachter/a407e9882ebe01520644cc9c257c8ec9/raw/8c837c52cfe22866ba4b49f2cf2f12336ca297d8/groups.csv
+
+        $path = $this->argument('path');
+
+        info('rogue:nassp-groups-import: Loading csv from ' . $path);
+
+        // Make a local copy of the CSV.
+        $temp = tempnam(sys_get_temp_dir(), 'command_csv');
+
+        file_put_contents($temp, fopen($this->argument('path'), 'r'));
+
+        // Load CSV contents.
+        $csv = Reader::createFromPath($temp, 'r');
+
+        $csv->setHeaderOffset(0);
+
+        $numImported = 0;
+        $numFailed = 0;
+        $numSkipped = 0;
+
+        $nhsGroupType = GroupType::firstOrCreate([
+            'name' => 'National Honor Society',
```

Comment:
I think we'll need a boolean attribute like `filter_by_state` on the `GroupType` model that we'd set to `true` here, to indicate that a Group Finder should first require user to select a `state` field to filter groups by. We won't always want to filter by state, for example -- March For Our Lives only has 230 groups, and their source dataset didn't contain `city` and `state` fields. Because it's only the 230 groups, it's preferable to have the finder filter just by the name field only.

I held off on adding this field to the `GroupType` model to keep this PR reviewable. But been pondering the field name, `filter_by_state`. It's more to indicate "hey we have valid state fields that would work better for users finding their group" but I don't like the names `has_state` or `has_location` much either.

Reply:
Makes sense to me! I like the explicit `filter_by_state` option better as well.